src/Components/Web.Extensions/src/InputFile/RemoteFileListEntryStream.cs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    // This class streams data from JS within the existing API limits of IJSRuntime.
+    // To produce good throughput, it prefetches up to buffer_size data from JS
+    // even when the consumer isn't asking for that much data, and does so by making
+    // N parallel requests in parallel (N ~= buffer_size / max_message_size).
+    //
+    // This should be understood as a TEMPORARY way to achieve the desired API and
+    // reasonable performance. Longer term we can surely replace this with something
+    // simpler and cleaner, either:
+    //
+    //  - Extending JS interop to allow streaming responses via SignalR's built-in
+    //    binary streaming support. That should reduce all of this to triviality.
+    //  - Or, failing that, at least use something like System.IO.Pipelines to manage
+    //    the supply/consumption of byte data with less custom code.
```

Comment:
It seems like this part of the comment no longer applies, since you have verified that `System.IO.Pipelines` isn't what we want here, and we don't have any concrete plans/timescales for implementing any way of passing unencoded byte arrays via `IJSRuntime`. So I'd suggest dropping it!

```suggestion
```